test/libsolidity/semanticTests/abiEncoderV2/abi_encode_v2_in_function_inherited_in_v1_contract.sol

```diff
@@ -30,6 +30,6 @@ contract C is B {
 }
 // ----
 // test() -> 77
-// gas irOptimized: 110731
+// gas irOptimized: 110325
```

Comment:
These gas differences probably come from the fact that `SolidityExecutionFramework` always used at least the sequence from `minimalStack`. Now all tests use default for unoptimized, which is `u` (unless the new setting is not `false`, which is never the case right now).

Reply:
Hm - but the gas data is ``irOptimized``, so only the ``--optimize`` run, isn't it? So different settings for the unoptimized run don't explain a gas difference.
But looking at the removed code in ``multiSourceCompileContract``: We did take ``m_compiler.yulIROptimized(contractName)`` out of the compiler stack, which already ran the Yul optimizer - and then fed it back to ``YulStack`` and optimized again, right? So previously, the optimizer was run twice for the tests, while now it's only run once? If that's correct, that'd explain the slight gas differences.

Reply:
Right. I looked at the removed `SolidityExecutionFramework` code too briefly before posting the PR and missed the fact that it was actually doing full `compile()` above the removed fragment and that fragment was only assembling.

Your analysis seems correct. We were optimizing twice by taking optimized output and running `optimize()` on the assembly stack again.

So, looks like before my removal of double optimization we were actually optimizing three times here :)